bluebutton-data-pipeline-fhir-load/src/main/java/gov/hhs/cms/bluebutton/datapipeline/fhir/transform/DataTransformer.java

```diff
@@ -916,32 +1001,163 @@ private TransformedBundle transformPartDEvent(RifRecordEvent<PartDEventRow> rifR
 
 		Bundle bundle = new Bundle();
 		ExplanationOfBenefit eob = new ExplanationOfBenefit();
-		eob.setId(IdType.newRandomUuid());
-		eob.addIdentifier().setSystem(CODING_SYSTEM_CCW_PDE_ID).setValue(record.partDEventId);
-		eob.setPatient(new Reference().setReference("Patient/" + record.beneficiaryId));
+		eob.setId("ExplanationOfBenefit/" + record.partDEventId);
+		// TODO Specify eob.type once STU3 is available (pharmacy)
+		Reference patientRef = referencePatient(record.beneficiaryId);
+		eob.setPatient(patientRef);
+		if (record.paymentDate.isPresent()) {
+			eob.setPaymentDate(Date.valueOf(record.paymentDate.get()));
+		}
 
 		ItemsComponent rxItem = eob.addItem();
 		rxItem.setSequence(1);
 		if (record.compoundCode == PartDEventRow.COMPOUND_CODE_COMPOUNDED) {
 			/* Pharmacy dispense invoice for a compound */
 			rxItem.setType(new Coding().setSystem(CODING_SYSTEM_FHIR_ACT).setCode("RXCINV"));
-		} else {
+		} else if (record.compoundCode == PartDEventRow.COMPOUND_CODE_NOT_COMPOUNDED) {
```

Comment:
I think you should probably have a fallthrough `else` clause here that throws a `BadCodeMonkeyException` if you encounter an unexpected value.